File: EngTaskGradingNetBESolution/EngTaskGradingNetBE/Models/Dtos/StudentDto.cs
using EngTaskGradingNetBE.Models.DbModel;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace EngTaskGradingNetBE.Models.Dtos
{
  public record StudentDto(
    int Id,
    string Number,
    string Email,
    string? Name,
    string? Surname,
    string? UserName,
    string? StudyProgram,
    string? StudyForm
  );

  public record StudentCreateDto(string Number, string Name, string Surname, string UserName, string Email, string StudyProgram, string StudyForm);
  public record StudentAnalysisResultDto(List<StudentCreateDto> Students, List<string> Errors);
}

File: eng-task-grading-react-fe/src/model/student-dto.ts
export interface StudentDto {
  id: number;
  number: string;
  email: string;
  name?: string;
  surname?: string;
  userName?: string;
  studyProgram?: string;
  studyForm?: string;
}

export interface StudentCreateDto {
  number: string;
  name: string;
  surname: string;
  userName: string;
  email: string;
  studyProgram: string;
  studyForm: string;
}

export interface StudentAnalysisResultDto {
  students: StudentCreateDto[];
  errors: string[];
}
